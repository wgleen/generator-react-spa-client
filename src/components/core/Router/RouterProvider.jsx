import React from 'react'
import PropTypes from 'prop-types'
import { BrowserRouter as Router } from 'react-router-dom'

const RouterProvider = (props) => {
  const { children } = props

  return <Router>{children}</Router>
}

RouterProvider.defaultProps = {
  children: null
}

RouterProvider.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.element,
    PropTypes.node,
    PropTypes.string
  ])
}

export default RouterProvider
