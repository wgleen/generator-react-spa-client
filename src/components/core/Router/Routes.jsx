import React, { lazy, Suspense } from 'react'
import { Route, Switch } from 'react-router-dom'
import routes from '../../../config/routes'

const Home = lazy(() => import('../../pages/Home/Home'))

const Routes = () => (
  <Suspense fallback="Loading">
    <Switch>
      <Route exact path={routes.root.path} component={Home} />
    </Switch>
  </Suspense>
)

export default Routes
