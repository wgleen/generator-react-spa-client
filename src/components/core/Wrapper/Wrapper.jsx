import React from 'react'
import PropTypes from 'prop-types'
import ReduxProvider from '../Redux/ReduxProvider'
import ThemeProvider from '../Theme/ThemeProvider'
import RouterProvider from '../Router/RouterProvider'

const Wrapper = (props) => {
  const { children } = props

  return (
    <>
      <ReduxProvider>
        <ThemeProvider>
          <RouterProvider>{children}</RouterProvider>
        </ThemeProvider>
      </ReduxProvider>
    </>
  )
}

Wrapper.defaultProps = {
  children: null
}

Wrapper.propTypes = {
  children: PropTypes.oneOfType([PropTypes.element, PropTypes.node])
}

export default Wrapper
