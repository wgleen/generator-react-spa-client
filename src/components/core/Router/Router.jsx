import React from 'react'
import RouterProvider from './RouterProvider'
import Routes from './Routes'

const Router = () => (
  <RouterProvider>
    <Routes />
  </RouterProvider>
)

export default Router
