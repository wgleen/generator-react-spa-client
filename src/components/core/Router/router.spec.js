import React from 'react'
import { mount } from 'enzyme'
import Wrapper from '../Wrapper/Wrapper'
import Router from './Router'

describe('<Router />', () => {
  it('should render correctly', () => {
    const component = mount(
      <Wrapper>
        <Router />
      </Wrapper>
    )

    expect(component).toMatchSnapshot()

    component.unmount()
  })

  it('should render the RouterProvider component containing the Routes component', () => {
    const component = mount(
      <Wrapper>
        <Router />
      </Wrapper>
    )

    expect(component.find('RouterProvider').find('Routes').exists()).toBeTrue()

    component.unmount()
  })
})
