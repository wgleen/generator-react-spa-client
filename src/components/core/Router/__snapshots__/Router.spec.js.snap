// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Router /> should render correctly 1`] = `
<Wrapper>
  <ReduxProvider>
    <Provider
      store={
        Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        }
      }
    >
      <ThemeProvider>
        <Component
          theme={
            Object {
              "colors": Object {
                "blue": "#2e7baf",
                "gray": "#737373",
                "grayDark": "#4d4d4d",
                "grayDarkest": "#0d0d0d",
                "grayLight": "#b3b3b3",
                "grayLightest": "#e5e5e5",
                "green": "#3cab3a",
                "primary": "#ec0000",
                "primaryDark": "#b90000",
                "primaryDarkest": "#860000",
                "primaryLight": "#ff8686",
                "primaryLightest": "#ffd2d2",
                "red": "#ff3225",
                "secondary": "#43bccd",
                "secondaryDark": "#2e9faf",
                "secondaryDarkest": "#247a86",
                "secondaryLight": "#94d9e2",
                "secondaryLightest": "#d0eef2",
                "yellow": "#e6e328",
              },
              "typography": Object {
                "fontFamilyPrimary": "Open Sans",
                "fontWeightPrimary": 300,
                "lineHeight": "normal",
                "textAlign": "left",
              },
            }
          }
        >
          <RouterProvider>
            <BrowserRouter>
              <Router
                history={
                  Object {
                    "action": "POP",
                    "block": [Function],
                    "createHref": [Function],
                    "go": [Function],
                    "goBack": [Function],
                    "goForward": [Function],
                    "length": 1,
                    "listen": [Function],
                    "location": Object {
                      "hash": "",
                      "pathname": "/",
                      "search": "",
                      "state": undefined,
                    },
                    "push": [Function],
                    "replace": [Function],
                  }
                }
              >
                <Router>
                  <RouterProvider>
                    <BrowserRouter>
                      <Router
                        history={
                          Object {
                            "action": "POP",
                            "block": [Function],
                            "createHref": [Function],
                            "go": [Function],
                            "goBack": [Function],
                            "goForward": [Function],
                            "length": 1,
                            "listen": [Function],
                            "location": Object {
                              "hash": "",
                              "pathname": "/",
                              "search": "",
                              "state": undefined,
                            },
                            "push": [Function],
                            "replace": [Function],
                          }
                        }
                      >
                        <Routes>
                          <Suspense
                            fallback="Loading"
                          >
                            <Switch>
                              <Route
                                component={
                                  Object {
                                    "$$typeof": Symbol(react.lazy),
                                    "_init": [Function],
                                    "_payload": Object {
                                      "_result": Promise {},
                                      "_status": 0,
                                    },
                                  }
                                }
                                computedMatch={
                                  Object {
                                    "isExact": true,
                                    "params": Object {},
                                    "path": "/",
                                    "url": "/",
                                  }
                                }
                                exact={true}
                                location={
                                  Object {
                                    "hash": "",
                                    "pathname": "/",
                                    "search": "",
                                    "state": undefined,
                                  }
                                }
                                path="/"
                              />
                            </Switch>
                            Loading
                          </Suspense>
                        </Routes>
                      </Router>
                    </BrowserRouter>
                  </RouterProvider>
                </Router>
              </Router>
            </BrowserRouter>
          </RouterProvider>
        </Component>
      </ThemeProvider>
    </Provider>
  </ReduxProvider>
</Wrapper>
`;
